spacy/lang/da/syntax_iterators.py

```diff
@@ -0,0 +1,77 @@
+# coding: utf8
+from __future__ import unicode_literals
+
+from ...symbols import NOUN, PROPN, PRON, VERB, AUX
+from ...errors import Errors
+
+
+def noun_chunks(doclike):
+    def is_verb_token(tok):
+        return tok.pos in [VERB, AUX]
+
+    def next_token(tok):
+        try:
+            return tok.nbor()
+        except IndexError:
+            return None
+
+    def get_left_bound(doc, root):
+        left_bound = root
+        for tok in reversed(list(root.lefts)):
+            if tok.dep in np_left_deps:
+                left_bound = tok
+        return left_bound
+
+    def get_right_bound(doc, root):
+        right_bound = root
+        for tok in root.rights:
+            if tok.dep in np_right_deps:
+                right = get_right_bound(doc, tok)
+                if list(
+                    filter(
+                        lambda t: is_verb_token(t) or t.dep in stop_deps,
+                        doc[root.i : right.i],
+                    )
+                ):
+                    break
+                else:
+                    right_bound = right
+        return right_bound
+
+    def get_bounds(doc, root):
+        return get_left_bound(doc, root), get_right_bound(doc, root)
+
+    doc = doclike.doc
+
+    if not doc.is_parsed:
+        raise ValueError(Errors.E029)
+
+    if not len(doc):
+        return
+
+    left_labels = [
+        "det",
+        "fixed",
+        "nmod:poss",
+        "amod",
+        "flat",
+        "goeswith",
+        "nummod",
+        "appos",
+    ]
+    right_labels = ["fixed", "nmod:poss", "amod", "flat", "goeswith", "nummod", "appos"]
+    stop_labels = ["punct"]
+
+    np_label = doc.vocab.strings.add("NP")
+    np_left_deps = [doc.vocab.strings.add(label) for label in left_labels]
+    np_right_deps = [doc.vocab.strings.add(label) for label in right_labels]
+    stop_deps = [doc.vocab.strings.add(label) for label in stop_labels]
+
+    for token in doclike:
+        if token.pos in [PROPN, NOUN, PRON]:
+            left, right = get_bounds(doc, token)
+            yield (left.i, right.i + 1, np_label)
+            token = right
```

Comment:
This final line doesn't affect the next token in the `for` loop like you're intending. There is more than one way to do this, but other languages typically keep track of the end of the previous noun chunk so they don't accidentally produce overlapping noun chunks. See `prev_end` in Swedish for an example: https://github.com/explosion/spaCy/blob/bc027dc35c7b2e4482561ca8a818a9b8904b4984/spacy/lang/sv/syntax_iterators.py#L31-L48

Reply:
Hi,
I chose to mimic the Spanish iterator because I find it better at finding all the noun chunks, but of course it makes sense to have non-overlapping chunks. 
I actually already had a loop for removing the nested chunks so I integrated it for this solution. Let me know if this solution works well for you.